dotnet/src/SemanticKernel.Abstractions/AI/ChatCompletion/ChatHistory.cs

```diff
@@ -201,6 +207,17 @@ public void RemoveRange(int index, int count)
         this._messages.RemoveRange(index, count);
     }
 
+    /// <summary>
+    /// An in-place enumeration of the chat history starting with the most recent message first.
+    /// </summary>
+    public IEnumerable<ChatMessageContent> ToDescending()
```

Comment:
We don't need to add utility methods to our public API surface, developers can implement things like this themselves and we can include implementations in our samples if needed

Reply:
This is a good change to zoom into, as I haven't provided any context.  This is an outgrowth of a new requirement I've discovered around chat history.  I can roll-it back for now and re-approach in an isolated PR.

Here is the core case to consider:

- Chat history is designed to always be in acending order (earliest message first).
- Common retrieval pattern for stored history it to retrieve in descending order (latest first).

As such, chat history should be able to support bi-directional enumeration without creating copies of arrays like the `Linq` extensions do (e.g. Reverse()).

I think this will be easier to grok in isolation...including now was opportunistic.